catalog/content/core/content-core-catalogerplugin/src/main/java/ddf/content/plugin/cataloger/CatalogContentPlugin.java

```diff
@@ -293,27 +303,36 @@ private Metacard generateMetacard(ContentItem contentItem, MimeType mimeType, St
 
             LOGGER.trace("EXITING: generateMetacard");
         } catch (IOException e) {
-            LOGGER.debug("Error encountered while using filed backed stream.", e);
+            LOGGER.debug("Error encountered while using file-backed stream.", e);
         }
 
-        if (properties != null) {
-            if (properties.containsKey(ContentMetacardType.RESOURCE_CHECKSUM)) {
-                String checksumValue = (String) properties
-                        .get(ContentMetacardType.RESOURCE_CHECKSUM);
-                contentMetacard.setAttribute(
-                        new AttributeImpl(ContentMetacardType.RESOURCE_CHECKSUM, checksumValue));
-            }
-
-            if (properties.containsKey(ContentMetacardType.RESOURCE_CHECKSUM_ALGORITHM)) {
-                String checksumAlgrotithm = (String) properties
-                        .get(ContentMetacardType.RESOURCE_CHECKSUM_ALGORITHM);
-                contentMetacard.setAttribute(
-                        new AttributeImpl(ContentMetacardType.RESOURCE_CHECKSUM_ALGORITHM,
-                                checksumAlgrotithm));
-            }
+        if (contentMetacard != null && properties != null) {
```

Comment:
I'm wondering if there's a better way to do this.  Right now anytime we add a plugin it looks like we have to add a `set<AttributeNameHere>()` method to account for that. Ideally we should just be able to drop in new plugins and things will get updated as appropriate.